docs/internal/new_product.rst

```diff
@@ -0,0 +1,333 @@
+.. highlight:: console
+.. internal_new_product
+
+=============================================
+Creating New Product
+=============================================
+
+Notes for adding new products to NCI datacube
+
+.. warning ::
+    This is still work in progress
+
+
+Test Database Access
+--------------------
+
+Make sure you have access to test database server that allows creation of new
+databases. Test database server is ``agdcdev-db.nci.org.au``. Configure your
+``~/.pgpass``, add new line like this one. Typically you should be able to
+re-use the same credentials as your production datacube.
+
+::
+
+    agdcdev-db.nci.org.au:6432:*:<your user name>:<your random password>
+
+Verify that things worked for you
+``psql -h agdcdev-db.nci.org.au -p 6432 datacube -l``,
+you should get a list of databases currently present.
+
+
+Preparing New Database
+----------------------
+
+It's best to create an empty database for testing new product. Test database
+should contain same metadata and products as the main database. Typically new
+product is going to be derived from existing products in the main database and
+since we track dataset lineage we need to have the same products in the test
+database as in the main.
+
+First create empty database
+
+.. code-block:: bash
```

Comment:
Yeah the problem is that I add extra line that runs the declared function, and Appendix is meant to be sourced without side-effects. Is there a way to combine `literalinclude` with extra text added?